src/Microsoft.TestPlatform.CrossPlatEngine/DataCollection/InProcDataCollectionExtensionManager.cs

```diff
@@ -189,6 +192,8 @@ private void InitializeInProcDataCollectors(string runSettings)
                     var interfaceTypeInfo = typeof(InProcDataCollection).GetTypeInfo();
                     foreach (var inProcDc in this.inProcDataCollectorSettingsCollection)
                     {
+                        inProcDc.CodeBase = Path.IsPathRooted(inProcDc.CodeBase) ? inProcDc.CodeBase : Path.Combine(defaultCodeBase, inProcDc.CodeBase);
+
                         var inProcDataCollector = this.CreateDataCollector(inProcDc, interfaceTypeInfo);
```

Comment:
Instead of passing inprocdc, pass individual settings like codebase, assemblyqualifiedname etc. so as to test it properly.